extras/scripts/postsubmit.bat

```diff
@@ -35,7 +35,7 @@ IF "%CMAKE_GENERATOR%"=="MinGW Makefiles" (
   mingw32-make -j12 || exit /b 1
 ) ELSE (
   type ALL_BUILD.vcxproj
-  msbuild ALL_BUILD.vcxproj || exit /b 1
+  msbuild ALL_BUILD.vcxproj /p:Configuration=%CONFIGURATION% || exit /b 1
```

Comment:
Hi, I'll take a closer look at the rest this weekend, but in the meantime:

What happens without the change to this file?
I'm wondering if this is a symptom that this change is not backwards compatible (i.e. if something that worked before wouldn't work anymore and people would have to change their build scripts to cope with it).

Reply:
This option specifies the setting on the Visual Studio IDE.
You can assume that it has the same effect as specifying the `CMAKE_BUILD_TYPE` variable in the CMake command.

If `/p:Configuration=(Debug|Release|RelWithDebInfo|MinSizeRe)` is not specified, the Debug configuration (`CMAKE_BUILD_TYPE=Debug`) is always built.

Reply:
Are you saying that before this PR these CI tests always used Debug for MSVC? 
Or that they would now but before they didn't?

If the former, this is a separate bugfix, so let's split it out in a 2nd PR to show that this PR works just with the other changes (and to check that with the same user-side flags we still build in debug when we need to and not in other cases).

If the latter, then if a user has a similar script for building Fruit (not just for ci) then it sounds like it would silently switch to building in debug mode unless they also change this, which would make this change not backwards compatible.

This PR enables new use cases that weren't possible before (which is great), but i want to make sure that it's not a breaking change.
I.e. Fruit users should not have to modify their build scripts after this, and we shouldn't change the behavior of any build scripts they might have had before, e.g. from building a release build to building a debug build).

Reply:
Before this PR, `set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}")` allowed the CMake multi-configuration generator to have only one configuration.

https://github.com/google/fruit/blob/8149131119475ccaae3c1ae386fe9f2c679317f0/CMakeLists.txt#L29-L30

Since there was only one configuration, it was adopted to the configuration specified by `CMAKE_BUILD_TYPE` when the `msbuild` command was used to build.
Therefore, this is not a bug fix, but an effect of this PR.
This may be affected if `fruit` was built by the `msbuild` command.

Nevertheless, Visual Studio IDE is described in the CMake documentation as a multi-configuration generator and the configuration must be determined at build time.
Unfortunately, if someone does not specify `/pConfiguration=<CMAKE_BUILD_TYPE>` in the `msbuild` command, then that command is also wrong.

For example, let's consider using the `msbuild` command to build `fruit` project and another CMake project that supports CMake multi-configuration generators.
First, generate `ALL_BUILD.vcxproj` with the `cmake -DCMAKE_BUILD_TYPE=Release` command, then build it with `msbuild ALL_BUILD.vcxproj`. `fruit` will be built with the Release configuration, but the other project will probably be built with the Debug configuration.
This result is caused by the `msbuild` command, which is a multi-configuration generator but does not specify the configuration at build time.
So, I do not believe it is necessary to maintain compatibility of the wrong commands.

Reply:
Whether or not we consider that a "bad" usage, it still makes this non-backwards compatible.
Which means that if I accept this PR as-is, the next release will have to be Fruit 4.0.0 to comply with semantic versioning, and then some users might be reluctant to upgrade fearing backwards compatibility issues.
Especially for a feature that AFAIK is only relevant to 1 platform (Windows) and 1 compiler (MSVC), since in other cases I don't think multi-configuration is used.

What if we use this new behavior only when `-DCMAKE_BUILD_TYPE=` is not set in the initial cmake command?
Setting that was required before, so there wouldn't be a backwards compatibility issue anymore.
And for users that do want to use the new multi-configuration mode, they'd probably not want to set a specific `-DCMAKE_BUILD_TYPE=` in the cmake command, right? (instead setting a configuration only in the msbuild command)
WDYT?

Or, if this wouldn't work, we could add a new CMake variable to opt-in to the new behavior, like `-DFRUIT_ENABLE_MULTI_CONFIGURATION=True` (exact name tbd).

Reply:
Not setting `CMAKE_BUILD_TYPE` (in the CMake configuration step) will cause a difference in the build flags that are expanded when generating test code, breaking compatibility.

Added opt-in option `FRUIT_ALLOW_MULTI_CONFIG` (default `OFF`).
For variable names, use "ALLOW" instead of "ENABLE", and "MULTI_CONFIG" since it follows the CMake variable `GENERATOR_IS_MULTI_CONFIG`.